gui/MainWindow.cxx

```diff
@@ -1201,15 +1201,7 @@ void MainWindowPrivate::loadImage(FrameData frame)
       this->UI.worldView->setImageData(imageData, size);
 
       // Update metadata view
-      if (!this->videoMetadataMap)
-      {
-        this->UI.metadata->updateMetadata(kv::metadata_vector{});
-      }
-      else
-      {
-        auto const& mdv = this->videoMetadataMap->get_vector(frame.id);
-        this->UI.metadata->updateMetadata(mdv);
-      }
+      this->UI.metadata->updateMetadata(videoSource->frame_metadata());
```

Comment:
Are there any performance implications to this change? I think this method (`frame_metadata`) should always return quickly?

Reply:
I did not observe any performances issues.  `frame_metadata` is pretty fast.  The data has already been decoded from the video, it just needs to be converted to KWIVER format on the fly.   I could consider using videoMetadataMap as a caching mechanism.  I just don't want to require it to be pre-populated before we can do anything else.  However, performance of this is not an issue currently, so I will probably leave it.

Reply:
> The data has already been decoded from the video

That's what I thought. If the format conversion isn't currently a pain point, it's probably fine. I was *expecting* it wouldn't be an issue, just asking for the sake of verification.